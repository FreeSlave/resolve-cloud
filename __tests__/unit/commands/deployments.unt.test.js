const mockData = [
  {
    deploymentId: 'deploymentId-1',
    applicationName: 'appname-1',
    lastUpdated: 'lastUpdated-1',
    url: 'url-1'
  },
  {
    deploymentId: 'deploymentId-2',
    applicationName: 'appname-2',
    lastUpdated: 'lastUpdated-2',
    url: 'url-2'
  }
]
const deployments = jest.fn().mockReturnValue(mockData)

jest.doMock('../../../utils/api', () => ({
  deployments
}))

const handler = require('../../../commands/deployments')

const infoSpy = jest.spyOn(console, 'info')
const errorSpy = jest.spyOn(console, 'error')

beforeEach(() => {
  deployments.mockClear()
  infoSpy.mockClear()
  errorSpy.mockClear()
})

describe('deployments', () => {
  test('default invoke', async () => {
    await handler()
    expect(deployments).toBeCalledWith()
    expect(infoSpy).toMatchSnapshot()
  })

  test('request failure', async () => {
    deployments.mockReturnValue(Promise.reject(Error('error')))

    await handler()
    expect(errorSpy).toBeCalledWith('error')
    expect(infoSpy).not.toBeCalled()
  })
})
