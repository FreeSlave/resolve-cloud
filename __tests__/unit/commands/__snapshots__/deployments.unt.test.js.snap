// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deployments default invoke 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "  ======= Deployments =======

  DeploymentId: deploymentId-1
  Application Name: appname-1
  Last Updated: lastUpdated-1
  URL: url-1
  

  DeploymentId: deploymentId-2
  Application Name: appname-2
  Last Updated: lastUpdated-2
  URL: url-2
  ",
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;
